wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -75,6 +75,34 @@ abstract class Wallet
     this
   }
 
+  private def utxosWithMissingTx: Future[Vector[SpendingInfoDb]] = {
+    for {
+      utxos <- spendingInfoDAO.findAllUnspent()
+      hasTxs <- FutureUtil.foldLeftAsync(Vector.empty[SpendingInfoDb], utxos) {
+        (accum, utxo) =>
+          // If we don't have tx in our transactionDAO, add it to the list
+          transactionDAO
+            .read(utxo.txid)
+            .map(txOpt => if (txOpt.isEmpty) accum :+ utxo else accum)
+      }
+    } yield hasTxs
+  }
+
+  protected def downloadMissingUtxos: Future[Unit] =
+    for {
+      utxos <- utxosWithMissingTx
+      blockHashes = utxos.flatMap(_.blockHash.map(_.flip))
+      // Download the block the tx is from so we process the block and subsequent txs
+      _ <- nodeApi.downloadBlocks(blockHashes.distinct)
+    } yield ()
+
+  override def start(): Future[Unit] = {
```

Comment:
This isn't what I had in mind, can't this go into the `WalletAppConfig`?

Reply:
No, because the `WalletAppConfig` doesn't have access to any of the DAOs or `nodeApi`

Reply:
:cry: , I guess if we do merge this though we can move the Thread inside of `AddressHandling.scala` into `Wallet.scala` and create/destroy it in the start/stop methods?